Assets/Python/Plague.py

```diff
@@ -41,97 +41,92 @@ def setup(self):
 		data.lGenericPlagueDates[3] = getTurnForYear(1850) + gc.getGame().getSorenRandNum(40, 'Variation') - 20
 
 		undoPlague = gc.getGame().getSorenRandNum(8, 'undo')
-		if (undoPlague <= 3):
+		if undoPlague <= 3:
 			data.lGenericPlagueDates[undoPlague] = -1
 
 
 	def checkTurn(self, iGameTurn):
 		if data.bNoPlagueOption:
 			return
 
-		for i in range(iNumTotalPlayersB):
-			if (gc.getPlayer(i).isAlive()):
-				if (data.players[i].iPlagueCountdown > 0):
-					data.players[i].iPlagueCountdown -= 1
-					if (data.players[i].iPlagueCountdown == 2):
-						self.preStopPlague(i)
-					elif (data.players[i].iPlagueCountdown == 0):
-						self.stopPlague(i)
-				elif (data.players[i].iPlagueCountdown < 0):
-					data.players[i].iPlagueCountdown += 1
-
-		for i in range(4):
-			if (iGameTurn == data.lGenericPlagueDates[i]):
+		for iPlayer in range(iNumTotalPlayersB):
+			if gc.getPlayer(iPlayer).isAlive():
+				if data.players[iPlayer].iPlagueCountdown > 0:
+					data.players[iPlayer].iPlagueCountdown -= 1
+					if data.players[iPlayer].iPlagueCountdown == 2:
+						self.preStopPlague(iPlayer)
+					elif data.players[iPlayer].iPlagueCountdown == 0:
+						self.stopPlague(iPlayer)
+				elif data.players[iPlayer].iPlagueCountdown < 0:
+					data.players[iPlayer].iPlagueCountdown += 1
+
+		for iPlague in range(4):

```

Comment:
`range(4) -> range(len(data.lGenericPlagueDates[iPlague]))`

Alternately:
`for iPlague, iPlagueDate in enumerate(data.lGenericPlagueDates):`
Then you can use `iPlagueDate` instead of `data.lGenericPlagueDates[iPlague]`